server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -111,6 +118,142 @@ private Optional<String> validateDateOfBirth(String dob) {
     return Optional.empty();
   }
 
+  private Form<EditTiClientInfoForm> validateFirstNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getFirstName())) {
+      return form.withError(FORM_FIELD_NAME_FIRST_NAME, "First name required");
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateLastNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getLastName())) {
+      return form.withError(FORM_FIELD_NAME_LAST_NAME, "Last name required");
+    }
+    return form;
+  }
+
+  private Boolean hasEmailChanged(String newEmail, AccountModel account) {
+    return !newEmail.equals(account.getEmailAddress());
+  }
+
+  private Form<EditTiClientInfoForm> validatePhoneNumber(Form<EditTiClientInfoForm> form) {
+    String phoneNumber = form.value().get().getPhoneNumber();
+    if (!Strings.isNullOrEmpty(phoneNumber)) {
+      phoneNumber = phoneNumber.replaceAll("[^0-9]", "");
+      if (phoneNumber.length() != 10) {
+        return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain 10 digits");
+      }
+      if (!phoneNumber.matches("[0-9]+")) {
```

Comment:
Can you please explain further?

Reply:
I think you've fixed this with the current patch!

What I meant was: If I passed in a phone number like "213945abc", then `phoneNumber.replaceAll("[^0-9]", "");` would change the string to be just "213945". Later, when the `if (!phoneNumber.matches("[0-9]+"))` check happens, the "abc" letters that I had included in the phone number were already removed, so that `if` check wasn't actually catching the letters.

The current implementation LGTM!